src/particles/wakefields/WakePush.cpp

```diff
@@ -0,0 +1,84 @@
+/* Copyright 2022-2023 The Regents of the University of California, through Lawrence
+ *           Berkeley National Laboratory (subject to receipt of any required
+ *           approvals from the U.S. Dept. of Energy). All rights reserved.
+ *
+ * This file is part of ImpactX.
+ *
+ * Authors: Alex Bojanich, Chad Mitchell, Axel Huebl
+ * License: BSD-3-Clause-LBNL
+ */
+#include "WakePush.H"
+
+#include <ablastr/particles/NodalFieldGather.H>
+
+#include <AMReX_BLProfiler.H>
+#include <AMReX_REAL.H>
+#include <AMReX_SPACE.H>
+
+
+namespace impactx::particles::wakefields
+{
+    void WakePush (
+        ImpactXParticleContainer & pc,
+        amrex::Gpu::DeviceVector<amrex::Real> const & convoluted_wakefield,
+        amrex::ParticleReal slice_ds,
+        amrex::Real bin_size,
+        amrex::Real bin_min,
+        int padding_factor
+    )
+    {
+        BL_PROFILE("impactx::particles::wakefields::WakePush")
+
+        using namespace amrex::literals;
+
+        // Loop over refinement levels
+        int const nLevel = pc.finestLevel();
+        for (int lev = 0; lev <= nLevel; ++lev)
+        {
+            // Loop over all particle boxes
+            using ParIt = ImpactXParticleContainer::iterator;
+
+#ifdef AMREX_USE_OMP
+#pragma omp parallel if (amrex::Gpu::notInLaunchRegion())
+#endif
+
+            for (ParIt pti(pc, lev); pti.isValid(); ++pti)
+            {
+                const int np = pti.numParticles();
+
+                // Physical constants and reference quantities
+                amrex::ParticleReal const mc_SI = pc.GetRefParticle().mass * (ablastr::constant::SI::c);
+                amrex::ParticleReal const pz_ref_SI = pc.GetRefParticle().beta_gamma() * mc_SI;
+
+                // Access data from StructOfArrays (soa)
+                auto& soa_real = pti.GetStructOfArrays().GetRealData();
+
+                amrex::ParticleReal* const AMREX_RESTRICT part_z = soa_real[RealSoA::z].dataPtr(); // Note: Currently for a fixed t
+                amrex::ParticleReal* const AMREX_RESTRICT part_pz = soa_real[RealSoA::pz].dataPtr(); // Note: Currently for a fixed t
```

Comment:
Is that comment correct or copied? I think we operate on fixed `s` here because we did no s-to-t transform.

Reply:
Chiming in here:  Yes, there should be no s-to-t transform.  It looks like this was copied.

Reply:
Let's then rename to:
```suggestion
                amrex::ParticleReal* const AMREX_RESTRICT part_t = soa_real[RealSoA::t].dataPtr();
                amrex::ParticleReal* const AMREX_RESTRICT part_pt = soa_real[RealSoA::pt].dataPtr();
```